### DISCUSSION TITLE
Can't map Mysql 'bit' type to bool

### DISCUSSION THREAD

**Discussion by qiuhuiming** [rbatis/rbatis#268] [General]
I have a student table in MySQL. There is a field  `is_administrator`. It is `bit` type. I want to map it to `Option<bool>`, but it failed.

ddl
```sql
create table student
(
    student_id       varchar(12)               not null
        primary key,
    is_administrator bit                       null,
);
```

entity defination:
```rust
#[crud_table]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Student {
    pub student_id: String,
    pub is_administrator: Option<bool>,
}
```

The error output:
```txt
thread 'repository::student::test::test_find_by_id_and_password' panicked at 'called `Result::unwrap()` on an `Err` value: E("invalid type: integer `1`, expected a boolean")', src/repository/student.rs:43:40
```

It works if I use `i32`  or `u8` type in rust. However, it looks weird. Is there any way to use `bool`?

**Comment by zhuxiujia** [comment]
> I have a student table in MySQL. There is a field `is_administrator`. It is `bit` type. I want to map it to `Option<bool>`, but it failed.
> 
> ddl
> 
> ```sql
> create table student
> (
>     student_id       varchar(12)               not null
>         primary key,
>     is_administrator bit                       null,
> );
> ```
> 
> entity defination:
> 
> ```rust
> #[crud_table]
> #[derive(Debug, Clone, Serialize, Deserialize)]
> pub struct Student {
>     pub student_id: String,
>     pub is_administrator: Option<bool>,
> }
> ```
> 
> The error output:
> 
> ```
> thread 'repository::student::test::test_find_by_id_and_password' panicked at 'called `Result::unwrap()` on an `Err` value: E("invalid type: integer `1`, expected a boolean")', src/repository/student.rs:43:40
> ```
> 
> It works if I use `i32` or `u8` type in rust. However, it looks weird. Is there any way to use `bool`?


mysql BIT is  rust type 'u8', 
but in bson is type 'i32'.

**Comment by Jzow** [comment]
In mysql, bit type corresponds to 0 and 1 of boolean type. In rust, u8 unsigned 8-byte type should be used for mapping
